lib/active_record/virtual_attributes/virtual_total.rb

```diff
@@ -69,24 +75,25 @@ def virtual_aggregate(name, relation, method_name = :sum, column = nil, options
         end
       end
 
+      def define_virtual_size_method(name, relation)
+        define_method(name) do
+          (has_attribute?(name) ? self[name] : send(relation).try(:size)) || 0
+        end
+      end
+
       def define_virtual_aggregate_method(name, relation, method_name, column)
-        if method_name == :size
-          define_method(name) do
-            (attribute_present?(name) ? self[name] : nil) || send(relation).try(:size) || 0
-          end
-        else
-          define_method(name) do
-            (attribute_present?(name) ? self[name] : nil) ||
-              begin
-                rel = send(relation)
-                if rel.loaded?
-                  rel.blank? ? nil : (rel.map { |t| t.send(column).to_i } || 0).send(method_name)
-                else
-                  # aggregates are not smart enough to handle virtual attributes
-                  arel_column = rel.klass.arel_attribute(column)
-                  rel.try(method_name, arel_column) || 0
-                end
-              end
+        define_method(name) do
+          if attribute_present?(name)
+            self[name]
+          else
+            rel = send(relation)
+            if rel.loaded?
+              rel.blank? ? nil : rel.map { |t| t.send(column) }.compact.send(method_name)
```

Comment:
this is the only line that changed in this method. I mentioned it in the opening comment.

Reply:
I removed this refactor